javalib/src/main/scala/java/util/AbstractMap.scala

```diff
@@ -0,0 +1,199 @@
+package java.util
+
+import scala.annotation.tailrec
+import scala.collection.JavaConversions._
+
+object AbstractMap {
+
+  private def entryEquals[K, V](entry: Map.Entry[K, V], other: Any): Boolean = {
+    other match {
+      case other: Map.Entry[_, _] =>
+        entry.getKey == other.getKey && entry.getValue == other.getValue
```

Comment:
really sorry for the overload of a JS operator, if there is a better idea I will for sure change it!